src/secp256k1.c

```diff
@@ -83,29 +83,43 @@ size_t secp256k1_context_prealloc_size(unsigned int flags) {
     return ret;
 }
 
-secp256k1_context* secp256k1_context_create(unsigned int flags) {
-    secp256k1_context* ret = (secp256k1_context*)checked_malloc(&default_error_callback, sizeof(secp256k1_context));
+secp256k1_context* secp256k1_context_prealloc_create(void* prealloc, unsigned int flags) {
```

Comment:
Needs a corresponding entry in `include/secp256k1.h`, and we should `EXPECT` that `prealloc` is non-NULL.